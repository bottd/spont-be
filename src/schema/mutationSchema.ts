import * as graphql from 'graphql';
import { createNewUser, getLocationByCoords, insertLocation } from './utils';
import { UserType, CoordType } from './types';

const {
  GraphQLObjectType,
  GraphQLString,
  GraphQLNonNull,
  GraphQLFloat,
} = graphql;

export const mutationSchema = new GraphQLObjectType({
  name: 'RootMutation',
  fields: {
    createUser: {
      type: UserType,
      args: {},
      resolve(parentValue, args) {
        return createNewUser();
      },
    },
    insertCoords: {
      type: CoordType,
      args: {
        userID: { type: GraphQLNonNull(GraphQLString) },
        latitude: { type: GraphQLNonNull(GraphQLFloat) },
        longitude: { type: GraphQLNonNull(GraphQLFloat) },
      },
      async resolve(parentValue, args) {
        const locations = await getLocationByCoords(args.latitude, args.longitude);
        if (locations.length > 1) {
          const join1 = await insertLocation(locations[0], {id: args.userID});
          const join2 = await insertLocation(locations[1], {id: args.userID});
          return join1;
        } else if (locations.length === 1) {
          const join1 = await insertLocation(locations[0], {id: args.userID});
          return join1;
        } else {
          return args;
        }
      },
    },
  },
});
