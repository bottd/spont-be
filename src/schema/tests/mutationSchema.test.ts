import { mutationSchema } from '../mutationSchema';

describe('mutationSchema', () => {
  it('should match snapshot', () => {
    expect(mutationSchema).toMatchSnapshot();
  });
});
