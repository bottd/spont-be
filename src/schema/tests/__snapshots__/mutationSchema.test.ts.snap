// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutationSchema should match snapshot 1`] = `"RootMutation"`;
